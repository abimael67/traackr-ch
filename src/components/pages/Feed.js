import { useState, useEffect } from 'react'
import Header from '../common/Header'
import NavBar from '../common/NavBar'
import InfluencerCard from '../common/InfluencerCard'
import getInfluencer from '../../api/handler'
import Pagination from '../common/Pagination'

const PAGINATION_SIZE = 10
const Feed = () => {
    const [currentPage, setCurrentPage] = useState(0)
    const [data, setData] = useState(null)
    const [keyword, setKeyword] = useState("")
    useEffect(() => setData(getInfluencer(currentPage*PAGINATION_SIZE, PAGINATION_SIZE, keyword)), [currentPage, keyword])

    const handleChange = (element) => {
        setKeyword(element.target.value)
        setCurrentPage(0)
    }
    const nextPage = () => {
        if(currentPage + 1 < Math.ceil(data.total / PAGINATION_SIZE))
            setCurrentPage(currentPage + 1)
    }
    const prevPage = () => {
        if(currentPage > 0)
            setCurrentPage(currentPage - 1)
    }

    const gotoPage = (page) => {
        setCurrentPage(page)
    }
    
    return (
        <div className="container">
            <Header />
            <NavBar handleChange={handleChange} />
            <div className="columns is-multiline">
                {data && data.data.map(influencer => <div key={influencer.uid} className="column is-half"> <InfluencerCard influencer={influencer} /></div>)}
            </div>
            <Pagination nextPage={nextPage} prevPage={prevPage} gotoPage={gotoPage} currentPage={currentPage} pagesCount={data ? Math.ceil(data.total / PAGINATION_SIZE) : 0} />
            <br/>
        </div>
    )
}

export default Feed