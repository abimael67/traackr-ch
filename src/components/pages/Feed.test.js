import renderer from 'react-test-renderer'
import Feed from './Feed'
import { render, fireEvent } from '@testing-library/react'
const test_influencers =
{
    "influencers": [
        {
            "audience": 2439571,
            "avatar": "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/834.jpg",
            "description": "Test description 1",
            "location": "Nampa, MS Dominican Republic",
            "name": "Test Name 1",
            "reach": 4.333,
            "resonance": 35.538,
            "uid": "036caeb1-0d56-46aa-b0ff-6a261a8d1239",
            "is_aligned": false,
            "channels": [
                {
                    "handle": "Giuseppe.Hickle54",
                    "platform": "INSTAGRAM"
                },
                {
                    "handle": "Giuseppe.Hickle54",
                    "platform": "TIKTOK"
                },
                {
                    "handle": "Giuseppe.Hickle54",
                    "platform": "GITHUB"
                },
                {
                    "handle": "Giuseppe.Hickle54",
                    "platform": "TWITCH"
                },
                {
                    "handle": "Giuseppe.Hickle54",
                    "platform": "FACEBOOK"
                },
                {
                    "handle": "Giuseppe.Hickle54",
                    "platform": "TWITTER"
                },
                {
                    "handle": "Giuseppe.Hickle54",
                    "platform": "REDDIT"
                }
            ]
        },
        {
            "audience": 268321,
            "avatar": "https://cloudflare-ipfs.com/ipfs/Qmd3W5DuhgHirLHGVixi6V76LhCkZUz6pnFt5AJBiyvHye/avatar/657.jpg",
            "description": "Test description 2",
            "location": "Federal Way, DE Uganda",
            "name": "Test Name 2",
            "reach": 96.75,
            "resonance": 57.816,
            "uid": "70d2ffe6-23cc-44b5-bec8-f822386710d5",
            "is_aligned": true,
            "channels": [
                {
                    "handle": "Herman_Dietrich70",
                    "platform": "TWITCH"
                },
                {
                    "handle": "Herman_Dietrich70",
                    "platform": "TIKTOK"
                },
                {
                    "handle": "Herman_Dietrich70",
                    "platform": "GITHUB"
                },
                {
                    "handle": "Herman_Dietrich70",
                    "platform": "TWITTER"
                },
                {
                    "handle": "Herman_Dietrich70",
                    "platform": "REDDIT"
                }
            ]
        },
    ]
}
//mock two sample influencers for consistency
jest.mock('../../api/data.json', () => test_influencers)

describe('Feed page tests', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<Feed />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('check search by name is working', () => {
        const feed = render(<Feed />)
        let influencerCards = feed.getAllByLabelText('influencer-card')
        expect(influencerCards).toHaveLength(2)
        const input = feed.getByLabelText('search-input')
        fireEvent.change(input, { target: { value: 'Test name 1' } })
        influencerCards = feed.getAllByLabelText('influencer-card')
        expect(influencerCards).toHaveLength(1)
    })

    it('check search by description is working', () => {
        const feed = render(<Feed />)
        let influencerCards = feed.getAllByLabelText('influencer-card')
        expect(influencerCards).toHaveLength(2)
        const input = feed.getByLabelText('search-input')
        fireEvent.change(input, { target: { value: 'Test description 2' } })
        influencerCards = feed.getAllByLabelText('influencer-card')
        expect(influencerCards).toHaveLength(1)
    })

})
