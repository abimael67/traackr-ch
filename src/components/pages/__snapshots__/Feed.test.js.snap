// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feed page tests renders correctly 1`] = `
<div
  className="container"
>
  <nav
    className="navbar"
  >
    <div
      className="navbar-brand"
    >
      <a
        className="navbar-item"
      >
        <img
          alt="Logo"
          src="logo.png"
        />
      </a>
    </div>
    <div
      className="navbar-menu"
    >
      <div
        className="navbar-end"
      >
        <div
          className="navbar-item"
        >
          <div
            className="field is-grouped"
          >
            <p
              className="control dash"
            >
              Dashboard
            </p>
            <p
              className="control"
            >
              <a
                className="button influencer-list-button"
              >
                <span>
                  Influencer List
                </span>
              </a>
            </p>
            <a>
              <ion-icon
                name="person-circle-outline"
                size="large"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>
  <nav
    className="level"
  >
    <div
      className="level-left"
    >
      <div
        className="level-item"
      >
        <div
          className="title"
        >
          <h1
            className="title is-1"
          >
            Influencers
          </h1>
        </div>
      </div>
    </div>
    <div
      className="level-right"
    >
      <div
        className="field"
      >
        <p
          className="control has-icons-left has-icons-right"
        >
          <input
            aria-label="search-input"
            className="input search-input"
            onChange={[Function]}
            placeholder="Search for an influencer"
            type="text"
          />
          <span
            className="icon is-small is-left"
          >
            <ion-icon
              name="search-outline"
            />
          </span>
        </p>
      </div>
    </div>
  </nav>
  <div
    className="columns is-multiline"
  />
  <nav
    aria-label="pagination"
    className="pagination is-centered"
    role="navigation"
  >
    <a
      className="pagination-previous"
      onClick={[Function]}
    >
      Previous
    </a>
    <a
      className="pagination-next"
      onClick={[Function]}
    >
      Next page
    </a>
    <ul
      className="pagination-list"
    />
  </nav>
  <br />
</div>
`;
